app/src/main/java/com/github/onlynotesswent/ui/overview/AddNote.kt

```diff
@@ -303,3 +281,37 @@ fun NoteDataTextField(
               focusedContainerColor = MaterialTheme.colorScheme.background,
               unfocusedContainerColor = MaterialTheme.colorScheme.background))
 }
+
+/**
+ * A composable function that displays the top app bar for the screen. It is composed of a back
+ * button and a title.
+ *
+ * @param title The title to be displayed in the top app bar.
+ * @param titleTestTag The test tag for the title.
+ * @param onBackClick The callback to be invoked when the back button is clicked.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ScreenTopBar(title: String, titleTestTag: String, onBackClick: () -> Unit) {
+  TopAppBar(
+      colors =
+          TopAppBarDefaults.topAppBarColors(containerColor = MaterialTheme.colorScheme.surface),
+      title = {
+        Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
+          Spacer(modifier = Modifier.weight(1f))
+          Text(
+              title,
+              color = MaterialTheme.colorScheme.onSurface,
+              modifier = Modifier.testTag(titleTestTag))
+          Spacer(modifier = Modifier.weight(2f))
+        }
+      },
+      navigationIcon = {
+        IconButton(onClick = onBackClick, Modifier.testTag("goBackButton")) {
+          Icon(
```

Comment:
Also for more modularity, we should have the option to define the Icon and not have its value hard coded

Reply:
Yeah, you're right, thanks